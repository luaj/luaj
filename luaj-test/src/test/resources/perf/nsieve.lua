-- The Computer Language Shootout
-- http://shootout.alioth.debian.org/
-- contributed by Isaac Gouy
-- modified by Mike Pall


local function nsieve(m,isPrime)
   for i=2,m do
      isPrime[i] = true
   end
   local count = 0

   for i=2,m do
      if isPrime[i] then
         for k=i+i, m, i do
            if isPrime[k] then isPrime[k] = false end
         end
         count = count + 1
      end
   end
   return count
end


local n = tonumber(arg and arg[1]) or 1
local flags = {}

local m = (2^n)*10000
print( string.format("Primes up to %8d %8d", m, nsieve(m,flags)))

m = (2^(n-1))*10000
print( string.format("Primes up to %8d %8d", m, nsieve(m,flags)))

m = (2^(n-2))*10000
print( string.format("Primes up to %8d %8d", m, nsieve(m,flags)))
