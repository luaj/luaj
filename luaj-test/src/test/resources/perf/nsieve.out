Primes up to    20000     2262
Primes up to    10000     1229
Primes up to     5000      669
